pkg/codegen/convert/plugin_mapper.go

```diff
@@ -216,24 +216,27 @@ func (l *pluginMapper) getMappingForPlugin(pluginSpec mapperPluginSpec, provider
 
 	// We'll delete this for loop once the plugins have had a chance to update.
 	for _, conversionKey := range conversionKeys {
-		data, mappedProvider, err := providerPlugin.GetMapping(conversionKey, provider)
+		mapping, err := providerPlugin.GetMapping(context.Background(), plugin.GetMappingRequest{
```

Comment:
This (and a lot of the other places) should probably be `context.TODO()` rather than Background. Background should only really be created in the top level of the program.

Reply:
Done. I have replaced `context.Background` with `context.TODO` in all non-test scenarios.